client/txn_test.go

```diff
@@ -288,7 +288,7 @@ func TestEndWriteRestartReadOnlyTransaction(t *testing.T) {
 			t.Errorf("expected error: %t, got error: %v", !success, err)
 		}
 		if !reflect.DeepEqual(expCalls, calls) {
-			t.Fatalf("expected %v, got %v", expCalls, calls)
+			t.Errorf("expected %v, got %v", expCalls, calls)
```

Comment:
this is a case of iteration over two test cases, isn't it? below, too, reads like iteration over several test cases, so failures are independent. Am I missing something?